scipy/stats/_morestats.py

```diff
@@ -3904,3 +3904,71 @@ def circstd(samples, high=2*pi, low=0, axis=None, nan_policy='propagate', *,
     if not normalize:
         res *= (high-low)/(2.*pi)  # [1] (2.3.14) w/ (2.3.7)
     return res
+
+
+@_axis_nan_policy_factory(
+    lambda x: x, result_to_tuple=lambda x: (x,), n_outputs=1
+)
+def directionalmean(samples, *, axis=0):
+    """
+    Computes the directional mean of a sample of vectors.
+
+    Serves as equivalent of the sample mean for directional data whose
+    magnitude is irrelevant, e. g. unit vectors. The vectorial data
```

Comment:
```suggestion
    magnitude is irrelevant, e. g. unit vectors. The vectorial data
```
Let's get some other opinions, but as I user I would probably expect this to normalize the vectors by default since it says magnitude is irrelevant. Perhaps we need a `normalize=True` option that can be turned off to save this expensive step.

Reply:
@rkern this `directionalmean` function computes... well, a directional mean given vectors. Currently, it expects the input vectors to be normalized to magnitude 1. Do you agree that the default should accept vectors of any magnitude and normalize them for the user? Maybe there should be a parameter `normalize` that turns off that step (when set False)?

Reply:
Yeah, that's probably wise. OTOH, if there were a handy function (not just a one-liner expression) that the user could call that would do the normalization, then I'd just show using that in the `Examples` instead.

Reply:
Added the `normalize` parameter and a corresponding test.